### PR TITLE
Add BSNL delivery detail

### PR DESCRIPTION
**Pull request by prabhanshuguptagit**
**Story card:** [sc-7624](https://app.shortcut.com/simpledotorg/story/7624/make-provider-specific-api-service-classes-bsnl)

## Because

We want to record communications and delivery results for BSNL SMSes.

## This addresses

Starts recording communications and delivery details from the BSNL channel. The detailable only contains the message ID on creation. The rest of the fields will be populated by a separate call to an API to fetch the message's status.

### CODE REVIEW

File: db/migrate/20220403123845_create_bsnl_delivery_details.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+class CreateBsnlDeliveryDetails < ActiveRecord::Migration[5.2]
+  def change
+    create_table :bsnl_delivery_details do |t|
+      t.string :message_id
+      t.string :message_status
+      t.string :result
```
Comment (harimohanraj89) [inline db/migrate/20220403123845_create_bsnl_delivery_details.rb:6]: If we wanna keep the PRs a bit scoped, wanna add just the fields we're using into this PR, and introduce the rest just-in-time? AKA the PR that populates them via API or consumes them somewhere?

**Review by harimohanraj89 — CHANGES_REQUESTED** [review]
Tiny things

File: db/migrate/20220403123845_create_bsnl_delivery_details.rb (hunk 1 above)
Comment (prabhanshuguptagit) [reply db/migrate/20220403123845_create_bsnl_delivery_details.rb:6]: That PR is coming right next up, so I'd like to keep it into a single migration for ease.

Comment (harimohanraj89) [reply db/migrate/20220403123845_create_bsnl_delivery_details.rb:6]: Makes sense if we have a very good picture of how this shakes out 👍

File: db/migrate/20220403123845_create_bsnl_delivery_details.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
+class CreateBsnlDeliveryDetails < ActiveRecord::Migration[5.2]
+  def change
+    create_table :bsnl_delivery_details do |t|
+      t.string :message_id
```
[the remark this answers is not in the document]
Comment (harimohanraj89) [reply db/migrate/20220403123845_create_bsnl_delivery_details.rb]: What about, say, dlt_template_id? Will we ever create BSNL details without a template ID?

**Review by harimohanraj89 — APPROVED** [review]

**Comment by harimohanraj89** [comment]
I've got one last question about more null constraints. PS we should back up the constraints with model validations. Definitely don't want to have to go all the way down to the DB to throw an (unfriendly) error.

Approving to unblock you once constraints and validations are all set!

**Pull request merged**